src/modules/emote_menu/components/Emotes.jsx

```diff
@@ -0,0 +1,121 @@
+import React, {useCallback, useEffect, useMemo, useRef, useState} from 'react';
+import Icon from 'rsuite/lib/Icon/index.js';
+import Button from 'rsuite/lib/Button/index.js';
+import VirtualizedList from './VirtualizedList.jsx';
+import emoteStore from '../stores/index.js';
+import styles from '../styles/emotes.module.css';
+import Emote from './Emote.jsx';
+
+const ROW_HEIGHT = 36;
+const WINDOW_HEIGHT = 308;
+const TOTAL_COLS = 7;
+
+function Emotes({onClick, onHover, section, onSection}) {
+  const wrapperRef = useRef(null);
+  const [, setUpdated] = useState(false);
+
+  const renderRow = useCallback(
+    ({key, style, index}) => {
+      const row = emoteStore.getRow(index);
+      return emoteStore.isHeader(index) ? (
+        <div key={key} style={style} className={styles.header}>
+          <Icon>{row.icon}</Icon>
+          {row.displayName}
+        </div>
+      ) : (
+        <div key={key} style={style} className={styles.row}>
+          {row.map((emote) => (
+            <Emote key={emote.code} emote={emote} onClick={() => onClick(emote)} onMouseOver={() => onHover(emote)} />
+          ))}
+        </div>
+      );
+    },
+    [onHover, onClick]
+  );
+
+  const handleHeaderChange = useCallback((index) => {
+    const header = emoteStore.getRow(index);
+    if (header != null) {
+      onSection(header.id);
+    }
+  });
+
+  useEffect(() => {
+    function callback() {
+      setUpdated((prev) => !prev);
+    }
+    emoteStore.on('updated', callback);
+
+    return () => {
+      emoteStore.off('updated', callback);
+    };
+  }, []);
+
+  useEffect(() => {
+    if (!section.scrollTo) return;
```

Comment:
nit: use a compare against null. `!= null`

Reply:
scrollTo is either `True` or `False`, the ambiguity is probably not ideal. It decides when the emotes list gets scrolled to on section change.